lib/membrane_h264_format/h264.ex

```diff
@@ -28,33 +28,25 @@ defmodule Membrane.H264 do
   @type framerate_t :: {frames :: pos_integer, seconds :: pos_integer}
 
   @typedoc """
-  Describes h264 stream format.
+  Describes h264 NALu format.
 
-  Byte-stream format (often reffered to as 'Annex B' because it is defined in Annex B
-  of [ITU-T H.264 Recommendation](http://www.itu.int/rec/T-REC-H.264-201704-I/en))
-  is suitable for writing to file or streaming with MPEG-TS.
+  `:annex_b` (defined in Annex B of [ITU-T H.264 Recommendation](http://www.itu.int/rec/T-REC-H.264-201704-I/en))
+  is suitable for writing to a file or streaming with MPEG-TS.
   In this format each NAL unit is preceded by three or four-byte start code (`0x(00)000001`)
   that helps to identify boundaries.
-
-  avc1 and avc3 are described by ISO/IEC 14496-15. In such stream NALUs lack the start codes,
-  but are preceded with their length. Avc streams are more suitable for placing in containers
-  (e.g. they are used by QuickTime (.mov), MP4, Matroska and FLV). Avc1 and avc3 differ in how PPS and SPS
-  (Picture Parameter Set and Sequence Parameter Set) are transported.
   """
-  @type stream_format_t :: :avc1 | :avc3 | :byte_stream
+  @type nalu_format_t :: :annex_b
```

Comment:
There are no use cases for `length_prefix` at the moment so I would add it once it is needed.

Reply:
If you really want to remove something unused, then I'd rather remove the whole type and struct key - there's no point in keeping a field with only one possible value. It's even worse that it's currently enforced.

However, I think I'd keep it anyway:
1. It's not problematic
2. It fulfils an informational role - you can learn something by reading the docs
3. We do use length-prefix format when using `ffmpeg`'s `h264` format muxer for test fixtures (although we do not use this struct then)
4. We actually could (or even should) use it [here, in MPEG-TS demuxer](https://github.com/membraneframework/membrane_mpegts_plugin/blob/master/lib/membrane_element_mpegts/demuxer.ex), as MPEG-TS uses a length-prefix format

Reply:
Btw. our documentation states that `:annex_b` `is suitable for writing to a file or streaming with MPEG-TS.`.

Is this fully correct? I mean, if MPEG-TS demuxer should use `:length_prefix` then the above statement is opposite to what we do?

Reply:
You're right, I've messed up the direction of https://ffmpeg.org/ffmpeg-bitstream-filters.html#h264_005fmp4toannexb - TS and `h264` file muxer use Annex B, which means start code prefix while MP4 uses length prefix mode.

So, the places where `:length_prefix` is really used is in https://github.com/membraneframework/membrane_flv_plugin/blob/master/examples/muxer.exs
 and https://github.com/membraneframework/membrane_matroska_plugin/blob/master/lib/membrane_matroska/muxer.ex
but we handle it by using `%MP4.Payload{content: %AVC1{}}` so I guess you were right about getting rid of it. Just ensure it is really removed, i.e. there's no field in the struct (I'm assuming we don't need the backward compatible version).

Plus I'd leave a note indicating this format uses start codes and length prefix is handled by `Membrane.MP4.Payload.AVC1`

Sorry for my mistake and creating confusion, hope this time my comment is valid :wink: